layout/shims.proj

```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Build">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+
+  <UsingTask TaskName="PrereleaseResolveNuGetPackageAssets" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <ItemGroup>
+    <!-- classic assembly refs -->
+    <ShimAssembly Include="mscorlib" />
+    <ShimAssembly Include="System" />
+    <ShimAssembly Include="System.Core" />
+    <ShimAssembly Include="System.Runtime.Serialization" />
+    <ShimAssembly Include="System.Xml" />
+    <ShimAssembly Include="System.Xml.Linq" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <RestoredRefPath>$(IntermediateOutputPath)CompatShims/ref/</RestoredRefPath>
+    <GenFacadesOutputPath>$(BinDir)CompatShims/facades/</GenFacadesOutputPath>
+    <NetCoreAppRefPath>$(LayoutBasePath)ref/netcoreapp1.1/</NetCoreAppRefPath>
+    <ProjectJson>$(MSBuildThisFileDirectory)project.json</ProjectJson>
+    <ProjectLockJson>$(MSBuildThisFileDirectory)project.lock.json</ProjectLockJson>
+  </PropertyGroup>
+
+  <Target Name="RestoreAndPublishForShims"
+          Inputs="$(ProjectJson)"
+          Outputs="$(ProjectLockJson);$(RestoredRefPath)">
+
+    <Exec Command="$(DnuRestoreCommand) $(ProjectJson)"
+          StandardOutputImportance="Low"
+          CustomErrorRegularExpression="(^Unable to locate .*)|(^Updating the invalid lock file with .*)" />
+
+    <PrereleaseResolveNuGetPackageAssets
+         IncludeFrameworkReferences="false"
+         NuGetPackagesDirectory="$(PackagesDir)"
+         ProjectLockFile="$(ProjectLockJson)"
+         TargetMonikers=".NETFramework,Version=v4.6.1">
+
+      <Output TaskParameter="ResolvedReferences" ItemName="ResolvedTargetingPackReference" />
+    </PrereleaseResolveNuGetPackageAssets>
+
+    <!-- Filter targeting pack down to just the set in ShimAssembly item group -->
+    <ItemGroup>
+      <ResolvedTargetingPackReferenceFilename Include="@(ResolvedTargetingPackReference -> '%(Filename)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </ResolvedTargetingPackReferenceFilename>
+      <ResolvedTargetingPackReferenceFilename Remove="@(ShimAssembly)" />
+      <PackageReferencesToRemove Include="@(ResolvedTargetingPackReferenceFilename -> '%(OriginalIdentity)')" />
+      <ResolvedTargetingPackReference Remove="@(PackageReferencesToRemove)" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(ResolvedTargetingPackReference)"
```

Comment:
For some reason I have in my head that depproj's only work for individual files but perhaps that is wrong. I will take a closer look.